src/models/utils.ts

```diff
@@ -516,3 +516,9 @@ export function deepClone<T>(obj: T): T {
 	});
 	return result;
 }
+
+/**
+ * Returns whether the credentials should use the native credential
```

Comment:
This comment doesn't make sense for something as generic as this - just because it's used for that purpose in one place doesn't mean it's the only way it's going to be used. I'd just remove the comment (the name alone is pretty clear what it means)